File: WorkoutTracker/src/app/exercises/models/exercise.ts
import { Equipment } from "src/app/equipments/models/equipment";
import { ChildMuscle } from "src/app/muscles/models/child-muscle";
import { WorkoutModel } from "src/app/shared/models/base/workout-model";
import { Workout } from "src/app/workouts/models/workout";
import { ExerciseType } from "./exercise-type";

export interface Exercise extends WorkoutModel {
    image: string|null;
    description: string|null;
    type: ExerciseType;
    isCreatedByUser: boolean;
    aliases: string[];

    equipments: Equipment[];
    workouts: Workout[];
    muscles: ChildMuscle[];
}

File: WorkoutTracker.Domain/Entities/Exercises/Exercise.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.Entities;
using WorkoutTracker.Domain.Entities.Exercises.ExerciseGroups;
using WorkoutTracker.Domain.Entities.Exercises.ExerciseSets;
using WorkoutTracker.Domain.Entities.Muscles;
using WorkoutTracker.Domain.Enums;

namespace WorkoutTracker.Domain.Entities.Exercises;

public class Exercise : BaseWorkoutModel
{
    public string? Image { get; set; }
    public string? Description { get; set; }
    public ExerciseType Type { get; set; }

    public string? CreatedByUserId { get; set; }

    public ICollection<Equipment> Equipments { get; set; } = null!;
    public ICollection<Muscle> WorkingMuscles { get; set; } = null!;

    public IEnumerable<ExerciseAlias>? ExerciseAliases { get; set; }
    public IEnumerable<ExerciseRecord>? ExerciseRecords { get; set; }
    public IEnumerable<ExerciseRecordGroup>? ExerciseRecordGroups { get; set; }
    public IEnumerable<ExerciseSetGroup>? ExerciseSetGroups { get; set; }
}
